odoo/addons/base/models/ir_mail_server.py

```diff
@@ -328,8 +328,17 @@ def build_email(self, email_from, email_to, subject, body, email_cc=None, email_
         msg['Message-Id'] = encode_header(message_id)
         if references:
             msg['references'] = encode_header(references)
+
+        force_smtp_from = self.env['ir.config_parameter'].sudo().get_param('mail.force.smtp.from')
```

Comment:
Shouldn't we take the `mail.catchall.domain` into account, so that the rewriting only takes place when the "From domain" is not the catchall domain? Or perhaps we can have 2 modes of operations to support:
- "dynamic rewriting": only if domain is foreign -> `notifications@mail.catchall.domain`
- "forced rewriting": always rewrite to fixed email From

The idea is to make it possible to avoid rewriting emails that would not cause spoofing issues, as that may significantly impact address books etc.

What do you think?

Reply:
Indeed we could support

  * odoo-style: rewrite only if not "@mail.company.com" as we own all mails linked to mail.company.com
  * gmail-style: rewrite as we don't own anything else than "robert.le.cuisiniste@gmail.com"

Reply:
@odony 
@tde-banana-odoo 

So we will have 2 system parameters ?

- `mail.force.smtp.from`, if set always force the FROM
- `mail.dynamic.smtp.from`, if set and previous not, rewrite only if the domain is not the `mail.catchall.domain`